feat(fxa-auth-server): Activate Mozilla account rebrand in fxa/subplat emails

vpomerleau opened the pull request:
## Because

* Firefox accounts/account will be renamed to Mozilla accounts/account starting Nov 1 2023

## This pull request

* Activate new l10n strings replacing Firefox accounts/account with Mozilla accounts/account
* Replace Firefox logo with Mozilla logo

## Issue that this pull request solves

Closes: #FXA-8321, #FXA-7987

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Can be reviewed with [storybook on stage](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/73396b0ddd9968554327a6c9c7109b5cab7bec5f/fxa-auth-server/index.html?path=/story/fxa-emails-layout--not-through-sync-flow)

dschom commented on packages/fxa-auth-server/lib/metrics/glean/server_events.ts:
```diff
@@ -65,7 +65,7 @@ class AccountsEventsServerEvent {
    * @param {string} utm_content - The content on which the user acted.  For example, if the user clicked on the "Get started here" link in "Looking for Firefox Sync? Get started here", then the value for this metric would be 'fx-sync-get-started'.  The value has a max length of 128 characters with the alphanumeric characters, _ (underscore), forward slash (/), . (period), % (percentage sign), an ...
    * @param {string} utm_medium - The "medium" on which the user acted.  For example, if the user clicked on a link in an email, then the value of this metric would be 'email'.  The value has a max length of 128 characters with the alphanumeric characters, _ (underscore), forward slash (/), . (period), % (percentage sign), and - (hyphen) in the allowed set of characters..
-   * @param {string} utm_source - The source from where the user started.  For example, if the user clicked on a link on the Firefox accounts web site, this value could be 'fx-website'.  The value has a max length of 128 characters with the alphanumeric characters, _ (underscore), forward slash (/), . (period), % (percentage sign), and - (hyphen) in the allowed set of characters..
+   * @param {string} utm_source - The source from where the user started.  For example, if the user clicked on a link on the Mozilla accounts web site, this value could be 'fx-website'.  The value has a max length of 128 characters with the alphanumeric characters, _ (underscore), forward slash (/), . (period), % (percentage sign), and - (hyphen) in the allowed set of characters..
```
Glad you updated the comments, and it's good to do so! Just incase you were wondering, updating comments at this moment in time isn't absolutely necessary. There are still going to be 'fxa' type references in comments and docs, and there will likely be another follow up for this at some point. It will become a hard requirement if ever move the git repo and/or swap domain.

dschom commented on packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml:
```diff
@@ -7,7 +7,7 @@
       <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
     </mj-raw>
-    <mj-title><%- subject %></mj-title>
+    <mj-title><%- locals.subject %></mj-title>
```
Was that just broken before you added `locals.`?

vpomerleau replied:
It doesn't look broken, but it was also the only spot where we didn't specify `locals.`... modified for consistency

dschom commented on packages/fxa-auth-server/lib/senders/emails/templates/verificationReminderFinal/index.txt:
```diff
@@ -1,4 +1,4 @@
-verificationReminderFinal-description = "A couple of weeks ago you created a Firefox account, but never confirmed it. For your security, we will delete the account if not verified in the next 24 hours."
+verificationReminderFinal-descriptio-2 = "A couple of weeks ago you created a Mozilla account, but never confirmed it. For your security, we will delete the account if not verified in the next 24 hours."
```
Should this be `description` instead of `descriptio`?

dschom commented:
@vpomerleau Looking good. Can you rebase on main to get the CSS fix for `fxa-auth-server` storybooks? Also checkout the comment about the typo. Then it's likely an r+ from me.

vpomerleau commented on packages/fxa-auth-server/lib/senders/emails/layouts/fxa/en.ftl:
```diff
@@ -8,6 +8,5 @@ fxa-header-mozilla-logo = <img data-l10n-name="mozilla-logo" alt="{ -brand-mozil
 fxa-header-sync-devices-image = <img data-l10n-name="sync-devices-image" alt="Sync devices">
 body-devices-image = <img data-l10n-name="devices-image" alt="Devices">
 fxa-privacy-url = { -brand-mozilla } Privacy Policy
-moz-accounts-privacy-url = { -product-mozilla-accounts(capitalization:"uppercase") } Privacy Policy
```
This new string is not needed, we can (and should) continue to use the existing `fxa-privacy-url` string.

vpomerleau commented:
@dschom Thanks for the review! I've addressed your comments in my latest push.

Also, as discussed on Slack, I reverted the ftl string removals and filed a separate ticket [FXA-8472](https://mozilla-hub.atlassian.net/browse/FXA-8472) to remove the old strings once the new branding is rolled out. The ticket includes a list of ftl ids that we'll want to remove.

Once the latest storybooks are deployed to stage and I've confirmed that styling is working (thanks @dschom for the fix!), I'll post in #moz-account-ux to get UX/UXC/SubPlat eyes on the updated emails.

vpomerleau commented on packages/fxa-auth-server/lib/senders/emails/templates/verificationReminderSecond/en.ftl:
```diff
@@ -4,6 +4,6 @@ verificationReminderSecond-title-3 = Don’t miss out on { -brand-mozilla }!
 verificationReminderSecond-description-3 = A few days ago you created a { -product-firefox-account }, but never confirmed it. Please confirm your account in the next 10 days or it will be automatically deleted.
 verificationReminderSecond-description-4 = A few days ago you created a { -product-mozilla-account }, but never confirmed it. Please confirm your account in the next 10 days or it will be automatically deleted.
 verificationReminderSecond-second-description = Your { -product-firefox-account } lets you sync your info across devices and unlocks access to more privacy-protecting products from { -brand-mozilla }.
-verificationReminderSecond-second-description-2 = Your { -product-mozilla-account } lets you sync your info across devices and unlocks access to more privacy-protecting products from { -brand-mozilla }.
```
@bcolsson this string was modified/added after your review. `verificationReminderSecond-second-description-2` has never been used in product, and an update was request by UXC. Just checking that this looks good!

bcolsson replied:
It's deleted now, but no issues with this string. 👍

vpomerleau commented on packages/fxa-auth-server/lib/senders/emails/templates/passwordResetAccountRecovery/includes.json:
```diff
@@ -4,7 +4,7 @@
   },
   "action": {
-    "id": "passwordResetAccountRecovery-action-2",
-    "message": "Create a new account recovery key"
```
@bcolsson this string was modified after your review based on UXC review of the emails. The original string was modified to avoid the text breaking over two lines in the button, but I mentioned that text would likely still break in other locales. FWIW, the text breaks between words even for German's extra long `Kontowiederherstellungsschlüssel erzeugen`

bcolsson replied:
Is there any way this could be improved?

vpomerleau replied:
I'm not sure. The copy was already reduced compared to the existing string, so I think this is an improvement, even for longer languages. I haven't seen any breakage _within_ words so this might be acceptable?  Unless we want to simplify even further and use "Create a (new) key", and that might not even help if locales need to use the full term for clarity.

Open to suggestions if you have them @bcolsson

bcolsson replied:
Oh, I assumed lines were breaking within words. If the button is sized to fit the German `Kontowiederherstellungsschlüssel` on one line without overflowing/breaking then I think that's okay.

dschom reviewed (approved):
LGTM! Thanks for the updates!
